packages/miniflare/src/workers/kv/namespace.worker.ts

```diff
@@ -0,0 +1,275 @@
+
+    // Get value from storage
+    validateGetOptions(key, { cacheTtl });
+    const entry = await this.storage.get(key);
+    if (entry === null) throw new HttpError(404, "Not Found");
+
+    // Return value in runtime-friendly format
+    const headers = new Headers();
+    if (entry.expiration !== undefined) {
+      headers.set(
+        KVHeaders.EXPIRATION,
+        millisToSeconds(entry.expiration).toString()
+      );
+    }
+    if (entry.metadata !== undefined) {
+      headers.set(KVHeaders.METADATA, JSON.stringify(entry.metadata));
+    }
+    return new Response(entry.value, { headers });
+  };
+
+  @PUT("/:key")
+  put: RouteHandler<KVParams> = async (req, params, url) => {
+    // Decode URL parameters
+    const key = decodeKey(params, url.searchParams);
+    const expirationParam =
+      url.searchParams.get(KVParams.EXPIRATION) ?? undefined;
+    const expirationTtlParam =
+      url.searchParams.get(KVParams.EXPIRATION_TTL) ?? undefined;
+
+    // Parse metadata if set
+    const metadataHeader = req.headers.get(KVHeaders.METADATA);
+    const metadata =
+      metadataHeader === null ? undefined : JSON.parse(metadataHeader);
+
+    validateKey(key);
+
+    // Normalise and validate expiration
+    const now = millisToSeconds(this.timers.now());
+    let expiration = normaliseInt(expirationParam);
+    const expirationTtl = normaliseInt(expirationTtlParam);
+    if (expirationTtl !== undefined) {
+      if (isNaN(expirationTtl) || expirationTtl <= 0) {
+        throw new HttpError(
+          400,
+          `Invalid ${KVParams.EXPIRATION_TTL} of ${expirationTtlParam}. Please specify integer greater than 0.`
+        );
+      }
+      if (expirationTtl < KVLimits.MIN_CACHE_TTL) {
+        throw new HttpError(
+          400,
+          `Invalid ${KVParams.EXPIRATION_TTL} of ${expirationTtlParam}. Expiration TTL must be at least ${KVLimits.MIN_CACHE_TTL}.`
+        );
+      }
+      expiration = now + expirationTtl;
+    } else if (expiration !== undefined) {
+      if (isNaN(expiration) || expiration <= now) {
+        throw new HttpError(
+          400,
+          `Invalid ${KVParams.EXPIRATION} of ${expirationParam}. Please specify integer greater than the current number of seconds since the UNIX epoch.`
+        );
+      }
+      if (expiration < now + KVLimits.MIN_CACHE_TTL) {
+        throw new HttpError(
+          400,
+          `Invalid ${KVParams.EXPIRATION} of ${expirationParam}. Expiration times must be at least ${KVLimits.MIN_CACHE_TTL} seconds in the future.`
+        );
+      }
+    }
+
+    // Validate metadata size
+    if (metadata !== undefined) {
+      assert(metadataHeader !== null);
+      const metadataLength = Buffer.byteLength(metadataHeader);
+      if (metadataLength > KVLimits.MAX_METADATA_SIZE) {
+        throw new HttpError(
+          413,
+          `Metadata length of ${metadataLength} exceeds limit of ${KVLimits.MAX_METADATA_SIZE}.`
+        );
+      }
```

Comment:
Similarly, I would expect this to be in `validator.worker.ts`. But I might be missing something